src/datamodel/quests/quest_new.rs

```diff
@@ -1,64 +1,45 @@
-use std::collections::HashMap;
-use std::rc::Rc;
+use std::hash::Hash;
 
-#[derive(Debug, PartialEq)]
+use crate::core::election::Election;
+
+#[derive(Hash, PartialEq, Eq, Clone, Debug, Copy)]
 pub enum Vote {
     Success,
     Failed,
 }
 
-pub struct QuestObjective {
-    pub num_players: u32,
-    pub required_successes: u32,
-}
-
-pub struct QuestNew {
-    members: HashMap<&'static str, bool>,
-    objective: Rc<QuestObjective>,
-    success_votes: u32,
+pub struct QuestNew<'a> {
+    election: Election<Vote>,
+    winner_rule: &'a (Vote, usize),
```

Comment:
Why is this a reference instead of having ownership of the winner rule here, or a `Rc`? I feel that borrowing the winner rule is strange, because it won't be usable outside of this struct while the instance lives.

Reply:
Not taking `Ownership` because we don't want `QuestNew` to consume `winner_rule` disabling it for the outer scope or forcing an unnecessary clonation.
Not an `Rc` because we don't need multiple ownership over `winner_rule`. Also, when `QuestNew` has finished the election it is going to be consume (drop) and `&winner_rule` will be destroyed.

Reply:
This sounds to me as if we should be using ownership then. If `winner_rule` is being borrowed by the `QuestNew` from the outer scope, then it won't be usable while it leaves. And why would we want to use the `winner_rule` for outside the `QuestNew` struct?

Reply:
To keep a record for example. I was thinking of this scenario.
But I think, it is not worth it and may be the best idea is to pass ownership to `QuestNew`.